superset-frontend/plugins/legacy-preset-chart-deckgl/src/layers/Contour/controlPanel.ts

```diff
@@ -0,0 +1,126 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ControlPanelConfig, sections } from '@superset-ui/chart-controls';
+import { t, validateNonEmpty } from '@superset-ui/core';
+import {
+  autozoom,
+  filterNulls,
+  jsColumns,
+  jsDataMutator,
+  jsOnclickHref,
+  jsTooltip,
+  mapboxStyle,
+  spatial,
+  viewport,
+} from '../../utilities/Shared_DeckGL';
+
+const config: ControlPanelConfig = {
+  controlPanelSections: [
+    sections.legacyRegularTime,
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        [spatial],
+        ['row_limit'],
+        ['size'],
+        [filterNulls],
+        ['adhoc_filters'],
+      ],
+    },
+    {
+      label: t('Map'),
+      expanded: true,
+      controlSetRows: [
+        [mapboxStyle, viewport],
+        [autozoom],
+        [
+          {
+            name: 'cellSize',
+            config: {
+              type: 'TextControl',
+              label: t('Cell Size'),
+              default: 300,
+              isInt: true,
+              description: t('The size of each cell in meters'),
+              renderTrigger: true,
+              clearable: false,
+            },
+          },
+        ],
+        [
+          {
+            name: 'aggregation',
+            config: {
+              type: 'SelectControl',
+              label: t('Aggregation'),
+              description: t(
+                'The function to use when aggregating points into groups',
+              ),
+              default: 'sum',
+              clearable: false,
+              renderTrigger: true,
+              choices: [
+                ['sum', t('sum')],
+                ['mean', t('mean')],
+              ],
+            },
+          },
+        ],
+        [
+          {
+            name: 'contours',
+            config: {
+              type: 'ContourControl',
+              label: t('Contours'),
+              renderTrigger: true,
+              description: t(
+                'Define contour layers. Isolines represent a collection of line segments that ' +
+                  'serparate the area above and below a given threshold. Isobands represent a ' +
+                  'collection of polygons that fill the are containing values in a given ' +
+                  'threshold range.',
+              ),
+            },
+          },
+        ],
+      ],
+    },
+    {
+      label: t('Advanced'),
+      controlSetRows: [
+        [jsColumns],
+        [jsDataMutator],
+        [jsTooltip],
+        [jsOnclickHref],
+      ],
+    },
+  ],
+  controlOverrides: {
+    size: {
+      label: t('Weight'),
+      description: t("Metric used as a weight for the grid's coloring"),
+      validators: [validateNonEmpty],
+    },
+  },
+  formDataOverrides: formData => ({
```

Comment:
Shouldn't that be 
```
formDataOverrides: formData => ({
    ...formData,
    size: getStandardizedControls().shiftMetric(),
  }),
```
?

It's used for retaining the control values when you switch the viz type, and it basically tells the control panel to prefill "size" control with the value of metric from the previous viz.